src/build_log.cc

```diff
@@ -151,10 +151,20 @@ bool BuildLog::RecordCommand(Edge* edge, int start_time, int end_time,
     Entries::iterator i = entries_.find(path);
     LogEntry* log_entry;
     if (i != entries_.end()) {
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
```

Comment:
create a helper function

```
template<typename TYPE_T>
TYPE_T * to_address(TYPE_T * ptr) { return ptr; }
#if CPP11
template<typename TYPE_T>
decltype(auto) to_address(std::unique_ptr<TYPE_T> const& ptr) { return ptr.get(); }
#endif 

So that you don't have all these ugly if-defs everywhere.